src/transformers/pipelines/text2text_generation.py

```diff
@@ -230,6 +242,22 @@ class TranslationPipeline(Text2TextGenerationPipeline):
 
     # Used in the return key of the pipeline.
     return_name = "translation"
+    src_lang: Optional[str] = None
+    tgt_lang: Optional[str] = None
+
+    def __init__(self, *args, src_lang=None, tgt_lang=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        if src_lang is not None:
+            self.src_lang = src_lang
+        if tgt_lang is not None:
+            self.tgt_lang = tgt_lang
+        if src_lang is None and tgt_lang is None:
+            task = kwargs.get("task", "")
+            tokens = task.split("_")
```

Comment:
nit - don't think `tokens` is a good name here -> `words` maybe or `items` => tokens makes me think about the tokenizers which are not related here at all IMO

Reply:
Fair enough